src/audio/module_adapter/module_adapter.c

```diff
@@ -730,12 +732,18 @@ int module_adapter_copy(struct comp_dev *dev)
 	}
 
 	if (mod->simple_copy) {
+		int mod_input = 0;
 		i = 0;
 		list_for_item(blist, &dev->bsource_list) {
-			comp_update_buffer_consume(source_c[i], mod->input_buffers[i].consumed);
+			/* some sources may be skipped, so skip_list must be used */
+			if (!(skip_list & BIT(i))) {
```

Comment:
Isn't this still problematic? What we need to do instead is iterate through 0 -> num_source_buffers and skip those with a different state isnt it?

Reply:
@ranj063 I'm open to better ways to do this, but wanted to keep this small to avoid conflicts with your #6836 (but I think we are anyways conflicting already). 

But. but. I think this is right. The module process sees "num_input_buffers" of input buffers and reports consumed on each of them. In this loop, we iteratate over these buffers using "mod_input". The "i" variable goes over all bsources. 

The consumed is updated in module_process() using the "mod->input_buffers, num_input_buffers" passed as inputs. Thus to pair right source with consumed value, we need to use "input_buffers[mod_input].consumed" as that's the buffer we passed for source "source_c[i]" and specific "i != mod_input" when any sources are skipped.

I can see values of 3+ million bytes passed as argument to comp_update_buffer_consume() when I run pause-resume tests on current main with current code and this PR fixes that. With asserts not enabled, this just silently leads to undefined behaviours.

Reply:
@ranj063 This is mimicking the loop on L650, but we can't do the state check here again as the state might have changed (unlikely but would be a very hard to debug problem), and rather we should stick to the decision we do on L655. If a bsource is skipped, we ignore it when calling the consume after process as well.